pkg/controllers/teamrbac/teamrolebinding_controller.go

```diff
@@ -99,50 +101,70 @@ func (r *TeamRoleBindingReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		switch len(teamRoleBinding.Spec.Namespaces) > 0 {
 		case true:
 			for _, namespace := range teamRoleBinding.Spec.Namespaces {
-				remoteObjectKey := types.NamespacedName{Name: teamRoleBinding.GetName(), Namespace: namespace}
+				remoteObjectKey := types.NamespacedName{Name: teamRoleBinding.GetRBACName(), Namespace: namespace}
 				remoteObject := &rbacv1.RoleBinding{}
 				err := remoteRestClient.Get(ctx, remoteObjectKey, remoteObject)
 				switch {
 				case apierrors.IsNotFound(err):
 					continue
 				case !apierrors.IsNotFound(err) && err != nil:
-					r.recorder.Eventf(teamRoleBinding, corev1.EventTypeWarning, greenhousev1alpha1.FailedDeleteRoleBindingReason, "Error retrieving RoleBinding %s for cluster %s in namespace %s", teamRoleBinding.GetName(), cluster.GetName(), namespace)
+					r.recorder.Eventf(teamRoleBinding, corev1.EventTypeWarning, greenhousev1alpha1.FailedDeleteRoleBindingReason, "Error retrieving RoleBinding %s for cluster %s in namespace %s", teamRoleBinding.GetRBACName(), cluster.GetName(), namespace)
 					// TODO: collect errors and return them
 					return ctrl.Result{}, err
 				default:
 					if err := remoteRestClient.Delete(ctx, remoteObject); err != nil {
-						r.recorder.Eventf(teamRoleBinding, corev1.EventTypeWarning, greenhousev1alpha1.FailedDeleteRoleBindingReason, "Error deleting RoleBinding %s for cluster %s in namespace %s", teamRoleBinding.GetName(), cluster.GetName(), namespace)
+						r.recorder.Eventf(teamRoleBinding, corev1.EventTypeWarning, greenhousev1alpha1.FailedDeleteRoleBindingReason, "Error deleting RoleBinding %s for cluster %s in namespace %s", teamRoleBinding.GetRBACName(), cluster.GetName(), namespace)
 						// TODO: collect errors and return them
 						return ctrl.Result{}, err
 					}
 				}
 			}
-			deletionSuccessful = true
+			rbacBindingsDeleted = true
 		case false:
-			remoteObjectKey := types.NamespacedName{Name: teamRoleBinding.GetName()}
+			remoteObjectKey := types.NamespacedName{Name: teamRoleBinding.GetRBACName()}
 			remoteObject := &rbacv1.ClusterRoleBinding{}
 			err := remoteRestClient.Get(ctx, remoteObjectKey, remoteObject)
-			switch {
-			case apierrors.IsNotFound(err):
-				deletionSuccessful = true
-			case err != nil:
-				r.recorder.Eventf(teamRoleBinding, corev1.EventTypeWarning, greenhousev1alpha1.FailedDeleteRoleBindingReason, "Error retrieving ClusterRoleBinding %s for cluster %s", teamRoleBinding.GetName, cluster.GetName())
+			if err != nil && !apierrors.IsNotFound(err) {
+				r.recorder.Eventf(teamRoleBinding, corev1.EventTypeWarning, greenhousev1alpha1.FailedDeleteRoleBindingReason, "Error retrieving ClusterRoleBinding %s for cluster %s", teamRoleBinding.GetRBACName(), cluster.GetName())
```

Comment:
We do not have any status/condition on the teamRoleBindings to report error to? Is this planned at some point?

Reply:
Yes that needs to be done: https://github.com/cloudoperators/greenhouse/issues/80